packages/@divvi/mobile/src/recaptcha/RecaptchaService.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Recaptcha,
+  RecaptchaAction,
+  type RecaptchaClient,
+} from '@google-cloud/recaptcha-enterprise-react-native'
+import { Platform } from 'react-native'
+import { getAppConfig } from 'src/appConfig'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'recaptcha/RecaptchaService'
+const RECAPTCHA_ACTION_TIMEOUT_MS = 10000 // 10 seconds
+
+export enum RecaptchaActionType {
+  PHONE_VERIFICATION = 'PHONE_VERIFICATION',
+  KEYLESS_BACKUP = 'KEYLESS_BACKUP',
+}
+
+class RecaptchaServiceImpl {
+  private client: RecaptchaClient | null = null
+  private initializationPromise: Promise<void> | null = null
+
+  async initializeIfNecessary(): Promise<void> {
+    if (this.client) return
+    if (this.initializationPromise) return this.initializationPromise
+
+    this.initializationPromise = this.init()
+    try {
+      await this.initializationPromise
+    } catch (err) {
+      // If initialization fails, clear the promise so that we can try again later
+      this.initializationPromise = null
+      throw err
+    }
+  }
+
+  private async init(): Promise<void> {
+    // If client is already initialized, no need to initialize again
+    if (this.client) {
+      return
+    }
+
+    Logger.info(TAG, 'Initializing reCAPTCHA client')
+    const clientKey = this.getSiteKey()
+    if (!clientKey) {
```

Comment:
nit: this duplicates logic in `isEnabled` (i.e., a falsey site key means not enabled)